pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1757,30 +1789,35 @@ func (service *SecurityPolicyService) getNamespaceUID(ns string) (nsUid types.UI
 	return namespace_uid
 }
 
-func (service *SecurityPolicyService) buildRulePortString(port *v1alpha1.SecurityPolicyPort) string {
+func (service *SecurityPolicyService) buildRulePortString(port *v1alpha1.SecurityPolicyPort, portNumber int) string {
 	protocol := string(port.Protocol)
-	if port.EndPort != 0 {
-		return fmt.Sprintf("%s.%s.%d", protocol, (port.Port).String(), port.EndPort)
+	// Build the rule port string name from port definition and not from a specific port number, this is a common case.
```

Comment:
yes, I had wanted to use portNumber == -1 to indicate the port is non named port. I feel that it might cause user confused. So, I introduce one more argument: hasNamedport to make the case more clear.